replica/database.cc

```diff
@@ -2122,7 +2118,8 @@ database::make_keyspace_config(const keyspace_metadata& ksm) {
     }
     cfg.enable_dangerous_direct_import_of_cassandra_counters = _cfg.enable_dangerous_direct_import_of_cassandra_counters();
     cfg.compaction_enforce_min_threshold = _cfg.compaction_enforce_min_threshold;
-    cfg.dirty_memory_manager = &_dirty_memory_manager;
+    // don't make system keyspace writes wait for user writes (if under pressure)
+    cfg.dirty_memory_manager = is_system ? &_system_dirty_memory_manager : &_dirty_memory_manager;
```

Comment:
Perhaps you could move the `if (extensions().is_extension_internal_keyspace(ksm->name()))` block here as well? This would be consistent with the spirit of the PR, I think.

Reply:
Good catch, it's indeed worth moving this if as well